myfitnesspal/client.py

```diff
@@ -939,42 +936,39 @@ def set_new_food(
             url,
             data,
         )
-        if result.status_code == 200:
-            document = lxml.html.document_fromstring(result.content.decode("utf-8"))
+        if not result.ok:
+            raise MyfitnesspalRequestFailed(
+                f"Request Error - Unable to submit food to MyFitnessPal: status code: {result.status_code}"
+            )
 
-            if not document.xpath(
-                # If list is empty there should be no error, could be replaced with assert
-                "//*[@id='errorExplanation']/ul/li"
-            ):
-                return True
-                # Would like to return FoodItem, but seems that it take
-                # to long until the submitted food is available in the DB
-                # return self.get_food_search_results("{} {}".format(brand, description))[0]
-            else:  # Error occurred
-                error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
-                error = error.replace("Description ", "")  # For cosmetic reasons
-                raise MyfitnesspalRequestFailed(
-                    f"Unable to submit food to MyFitnessPal: {error}"
-                )
+        document = lxml.html.document_fromstring(result.content.decode("utf-8"))
 
-        elif not result.ok:
+        if document.xpath(
+            # If list is empty there should be no error, could be replaced with assert
+            "//*[@id='errorExplanation']/ul/li"
+        ):
+            error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
+            error = error.replace("Description ", "")  # For cosmetic reasons
             raise MyfitnesspalRequestFailed(
-                f"Request Error - Unable to submit food to MyFitnessPal: status code: {result.status_code}"
+                f"Unable to submit food to MyFitnessPal: {error}"
             )
 
+        # Would like to return FoodItem, but seems that it take
+        # to long until the submitted food is available in the DB
+        # return self.get_food_search_results("{} {}".format(brand, description))[0]
+
     def set_new_goal(
         self,
-        energy: float = None,
-        energy_unit: str = None,
+        energy: float,
```

Comment:
The linter was complaining to me that the energy value could be `None`  here: https://github.com/coddingtonbear/python-myfitnesspal/pull/121/commits/419c4163da17409be8c52824837abcb53ec8cd05#diff-30f1e1808bae1ffacf52afff3d5e9489e4224b19dfd67f94653af05fcfbf09e0L1095, and when looking through the method it looked like we were always raising an exception if `energy` were not a `float`; so I've here just changed the signature to demand that you provide an energy value and was then able to remove the logic asserting that that value was non-`None`.

Reply:
Jep, but now the energy argument is mandatory and this breaks various situations of the function based on the provided arguments. As mentioned in the docstrings the energy argument was not needed if all macro or percentage values for each macro were provided.. Therefore, I only raised an exception for the missing energy value in if-branches where we already checked if the other arguments are missing too. A possible solution would be to set the default value for energy to -1 and replace the None-Checks with energy < 0?

Reply:
I might be missing one of the branches, but it looks like all logic branches here required `energy`, here, no? 

https://github.com/Dnic94/python-myfitnesspal/compare/2f76e2f9fda53a835c79b61dbe5f53fa6447b7e9...80878cf#diff-30f1e1808bae1ffacf52afff3d5e9489e4224b19dfd67f94653af05fcfbf09e0R1020

Reply:
Unfortunately, your links seem not to work.

In the else-branch starting line 1072 (commit 419c4163da17409be8c52824837abcb53ec8cd05), the energy value was not needed. I also mentioned it in a comment there, line 1076.

But the value will be corrected anyways if it is not plausible to the passed macros. Now, user can pass “0” for energy or something like that if they want to use this functionality. 
So, an extra hint in the docstrings should be fine and it makes the code much cleaner.

Reply:
Looking at that else branch, it looks like if you were to pass `None` (the default), you'd get a `TypeError` because of this line: https://github.com/coddingtonbear/python-myfitnesspal/commit/419c4163da17409be8c52824837abcb53ec8cd05#diff-30f1e1808bae1ffacf52afff3d5e9489e4224b19dfd67f94653af05fcfbf09e0R1077, no?

Reply:
Wow, I thought I tested all cases, but seems like it was not the case...
So, nevermind 😅 🤦‍♂️

Reply:
It happens to all of us!  It's also one of the reasons why unit tests are so popular as a bug-reduction technique.

So, regarding approaches -- I'm not exactly sure what direction to go, but as long as `mypy` isn't complaining about types, I'm pretty alright with any path forward.

Reply:
Here is nothing left to adjust. I am fine with your changes and I already adjusted the docstrings in the last commit.
So I think we can close this conversation. 👍